pkg/webhook/webhook.go

```diff
@@ -644,17 +661,8 @@ func MutateHandler(w http.ResponseWriter, req *http.Request) {
 				resourceList[corev1.ResourceName(name)] = *resource.NewQuantity(number, resource.DecimalSI)
 			}
 
-			for resource, quantity := range resourceList {
-				patch = append(patch, jsonPatchOperation{
-					Operation: "add",
-					Path:      "/spec/containers/0/resources/requests/" + toSafeJsonPatchKey(resource.String()), // NOTE: in future we may want to patch specific container (not always the first one)
-					Value:     quantity,
-				})
-				patch = append(patch, jsonPatchOperation{
-					Operation: "add",
-					Path:      "/spec/containers/0/resources/limits/" + toSafeJsonPatchKey(resource.String()),
-					Value:     quantity,
-				})
+			if v, exists := pod.ObjectMeta.Labels[InjectResourceNameLabel]; !exists || v != "false" {
```

Comment:
doesn't it break when NAD contains `k8s.v1.cni.cncf.io/resourceName` and pod has label with nri.npwg.org/resourcename: false, this would make device not get injected by device plugin and deviceID not set in the subsequent CNI cmdAdd invocation. or should there be a check here not to have both settings together ?

Reply:
Yes, this is intentional. It allows user to skip network resource injection when NRI is running.
The issue it addresses is describe here: https://github.com/k8snetworkplumbingwg/network-resources-injector/issues/56

Reply:
@zshi-redhat ok, but doesn't this make CNI to behave in unexpected way as deviceID is not passed (for example, in ovs-cni attaches veth device into pod container instead of sriov VF, host-device CNI might throw an error which lead pod into container creating error state), are we also fine with not exposing which VF assigned to pod container (atleast via PCI Address env variable from device plugin) ? I'm not sure if I'm missing something here.

Reply:
By skipping the resource injection in NRI, user will need to explicitly define the resource request/limit in pod spec, because NRI is not smart enough to know which container it should inject resource to. 
The default behavior is to always inject resource name as long as it is found in net-attach-def, this is not changed.

Reply:
ok, I understand it now, should we add a check if pod spec contains resource request/limit in any one of the containers when net-attach-def has resource name and resource injection is disabled in pod spec ?

Reply:
I think that's a good idea.
We probably also want to rename the current honorExistingResources flag to something else, because honorExistingResources sounds like it is to disable resource injection when resource is found in pod spec. wdyt?

Reply:
@zshi-redhat now the changes look fine to me by reusing the existing honorExistingResources flag and remove the extra resource requests/limits as directed by networks annotation.